pkg/registry/flowcontrol/ensurer/flowschema.go

```diff
@@ -0,0 +1,213 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ensurer
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	flowcontrolv1beta1 "k8s.io/api/flowcontrol/v1beta1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	flowcontrolclient "k8s.io/client-go/kubernetes/typed/flowcontrol/v1beta1"
+	flowcontrolapisv1beta1 "k8s.io/kubernetes/pkg/apis/flowcontrol/v1beta1"
+)
+
+var (
+	errObjectNotFlowSchema = errors.New("object is not a FlowSchema type")
+)
+
+// FlowSchemaEnsurer ensures the specified bootstrap configuration objects
+type FlowSchemaEnsurer interface {
+	Ensure([]*flowcontrolv1beta1.FlowSchema) error
+}
+
+// FlowSchemaRemover removes the specified bootstrap configuration objects
+type FlowSchemaRemover interface {
+	Remove([]*flowcontrolv1beta1.FlowSchema) error
```

Comment:
I should have caught that before - sorry.

For consistency with client-go semantics (and its Delete method), can we actually just do:
```
Remove([]string) error
```

i.e. pass only the names of FlowSchemas (same for PL)


It would allow to also simplify a bit some code.

Reply:
yes, if we use `[]string` that will eliminate the type conversion code.